grunt/service-worker-fetch-listener.js

```diff
@@ -0,0 +1,61 @@
+self.addEventListener('fetch', function(event) {
+  if (event.request.method === 'GET') {
+    // The `if` condition below should be handled on the line above i.e.
+    // if (event.request.method === 'GET' && event.request.headers.get('accept').indexOf('application/json') === -1) {
+    // But it fails when running e2e test
+    // this has been attributed by the root url serving both json GET request and the cached HTML page
+    if (event.request.headers.get('accept').indexOf('application/json') !== -1) {
```

Comment:
Any dummy inline code on how we can achieve this?

Reply:
Here is how you set the template file path. https://github.com/GoogleChromeLabs/sw-precache#templatefilepath-string

Current we are using this default: https://github.com/GoogleChromeLabs/sw-precache/blob/master/service-worker.tmpl

So we'd have to fork that file, check it in, set our build logic to use it, and then alter it directly instead of appending to it.

Reply:
Oh nevermind. You did it already!